cli/command/registry/login.go

```diff
@@ -96,7 +104,37 @@ func runLogin(dockerCli command.Cli, opts loginOptions) error {
 		authConfig.Password = ""
 		authConfig.IdentityToken = response.IdentityToken
 	}
-	if err := dockerCli.ConfigFile().GetCredentialsStore(serverAddress).Store(authConfig); err != nil {
+
+	creds := dockerCli.ConfigFile().GetCredentialsStore(serverAddress)
+
+	store, isDefault := creds.(*credentials.FileStore)
+	if isDefault {
+		fmt.Fprintln(dockerCli.Err(), fmt.Sprintf(unencryptedWarning, store.GetFilename()))
+
+		// If this is not a terminal, let's assume they want it
+		// (sadly), because people may have been scripting insecure
+		// logins and we don't want to break them. Maybe they'll see
+		// the warning in their logs and fix things. If it is
+		// interactive, let's ask them if they really want to do this.
+		if dockerCli.In().IsTerminal() {
```

Comment:
Please extract the entire confirmation block into a new function so the comment can be a godoc string instead of inline comment.